executor/sort.go

```diff
@@ -119,85 +115,136 @@ func (e *SortExec) Open(ctx context.Context) error {
 	}
 	e.exceeded = 0
 	e.spilled = 0
-	e.rowChunksInDisk = nil
-	e.rowPtrsInDisk = e.rowPtrsInDisk[:0]
 	e.partitionList = e.partitionList[:0]
-	e.partitionRowPtrs = e.partitionRowPtrs[:0]
 	return e.children[0].Open(ctx)
 }
 
 // Next implements the Executor Next interface.
+// Sort constructs the result following these step:
+// 1. Read as mush as rows into memory.
+// 2. If memory quota is triggered, sort these rows in memory and put them into disk as partition 1, then reset
+//    the memory quota trigger and return to step 1
+// 3. If memory quota is not triggered and child is consumed, sort these rows in memory as partition N.
+// 4. Merge sort if the count of partitions is larger than 1. If there is only one partition in step 4, it works
```

Comment:
Could you add a benchmark for memory sort, to make sure that it does not has performance regression?
Also, you can provide a disk-based sort compared with memory sort.